flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -1576,4 +1576,15 @@ public ExecutionDeploymentListener getExecutionDeploymentListener() {
     public boolean isDynamic() {
         return isDynamic;
     }
+
+    @Override
+    public Optional<ExecutionVertexID> getExecutionVertexId(ExecutionAttemptID id) {
+        Execution execution = this.getRegisteredExecutions().get(id);
+        return Optional.ofNullable(execution).map(Execution::getVertex).map(ExecutionVertex::getID);
+    }
+
+    @Override
+    public Optional<ExecutionVertex> getExecutionVertex(final ExecutionVertexID executionVertexId) {
```

Comment:
We already have `DefaultExecutionGraph#getExecutionVertexOrThrow` in place, so we should avoid adding a new method

Reply:
But that could be handled in the method `convertFailures` itself, when the optional is empty